From:           Alexander Osthof <aosthof@suse.de>
Subject:        Add message: No mptable found

If no mptable is found better tell user about it.
This also sticks to x86_64 implementation now.

Patch-mainline:
References:     142454

Acked-by:
Signed-off-by:  aosthof@suse.de

 arch/i386/kernel/mpparse.c |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

Index: linux-2.6.15/arch/i386/kernel/mpparse.c
===================================================================
--- linux-2.6.15.orig/arch/i386/kernel/mpparse.c
+++ linux-2.6.15/arch/i386/kernel/mpparse.c
@@ -824,8 +824,14 @@ void __init find_smp_config (void)
 	 */
 
 	address = get_bios_ebda();
-	if (address)
-		smp_scan_config(address, 0x400);
+	if (address) {
+		/*
+		 * If this call of smp_scan_config() also returns 0 like the
+		 * three calls before, no mptable could be found at all.
+		 */
+		if (!smp_scan_config(address, 0x400))
+			printk(KERN_INFO "No mptable found.\n");
+	}
 }
 
 /* --------------------------------------------------------------------------
